### PR TITLE
Payee 306 fe upgrade cy tax free amount

### PR DESCRIPTION
**Pull request by mahowarth**
- [x] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/views/TaxFreeAmountView.scala.html [hunk 1]
```diff
@@ -21,111 +21,122 @@
...
-                    <p id="ledeMessage" class="flush--top">
-                        <span class="lede">@Messages("tai.taxFreeAmount.summarysection.taxFreeExplanation")</span>
+            <div id="taxFreeAmountSummary">
+                    <p class="govuk-body-l">
+                        @Messages("tai.taxFreeAmount.summarysection.taxFreeExplanation")
                     </p>
 
-                    <h2 class="flush--bottom">@Messages("tai.taxFreeAmount.summarysection.heading")
-                        <span class="display-block bold-large section section--narrow flush--bottom">@viewModel.annualTaxFreeAmount</span>
+                    <h2 id="xyz" class="govuk-heading-m">
```
Comment (sergi-hmrc) [inline app/views/TaxFreeAmountView.scala.html:59]: id xyz?

Comment (masahmad) [reply app/views/TaxFreeAmountView.scala.html:59]: oops

Comment (masahmad) [reply app/views/TaxFreeAmountView.scala.html:59]: id="xyz" can be removed Maria -  I must have put it in by mistake

File: app/assets/stylesheets/modules/_cya-tax-summary.scss [hunk 2]
```diff
@@ -13,6 +13,17 @@
 
 }
 
+div.tai-tax-summary-table:nth-last-of-type(1) dl {
```
Comment (vivekhmrc) [inline app/assets/stylesheets/modules/_cya-tax-summary.scss:16]: Not sure about this part.Rest all looks good

**Review by vivekhmrc — APPROVED** [review]
LGTM 👍  other than one usage of div

File: app/assets/stylesheets/modules/_cya-tax-summary.scss (hunk 2 above)
Comment (masahmad) [reply app/assets/stylesheets/modules/_cya-tax-summary.scss:16]: its basically a hack  to make it look like the prototype - remember yesterday in retro I mentioned the prototype had a table but staging we use a summary list component and we kind of ended up with a hybrid solution where we were formatting the summary list so it looks like the table in the prototype.  The use of a Table in the prototype is incorrect in my opinion

File: test/views/html/TaxFreeAmountViewSpec.scala [hunk 3]
```diff
@@ -37,9 +37,9 @@ class TaxFreeAmountViewSpec extends TaiViewSpec {
         doc must haveDivWithId("taxFreeAmountSummary")
         doc must haveElementAtPathWithText(
           "div[id=taxFreeAmountSummary]>h2",
-          s"${messages("tai.taxFreeAmount.summarysection.heading")} £2020")
+          messages("tai.taxFreeAmount.summarysection.heading"))
```
Comment (brendanstraughan) [inline test/views/html/TaxFreeAmountViewSpec.scala:40]: should "2020" have been removed here?

File: app/views/includes/taxFreeAmountSummary.scala.html [hunk 4]
```diff
@@ -65,16 +65,16 @@ <h3 class="heading-small section--narrow flush" id="summaryTable@{itemIdx.toStri
...
                                 }}
                             }
                         </div>
                     }
                 </dl>
                 } else {@if(item.rows.length == 1) {
 
-                <dl id="summaryTable@{itemIdx.toString}Body" class="govuk-check-your-answers cya-questions-long govuk-check-your-answers--tax-summary @if(itemIdx == items.length && item.rows.length == 1){subsection highlight heading-small} else {flush}">
-                    <div id="summaryTable@{itemIdx.toString}Row1">
+                <dl id="summaryTable@{itemIdx.toString}Body" class="govuk-summary-list govuk-check-your-answers cya-questions-long govuk-check-your-answers--tax-summary @if(itemIdx == items.length && item.rows.length == 1){subsection xxxhighlight heading-small} else {flush}">
```
Comment (brendanstraughan) [inline app/views/includes/taxFreeAmountSummary.scala.html:76]: is xxxhighlight correct? Never seen it before

File: app/views/TaxFreeAmountView.scala.html [hunk 5]
```diff
@@ -21,111 +21,122 @@
...
+                    <h2 class="govuk-heading-m">@Messages("tai.taxFreeAmount.detailsection.heading")</h2>
 
-                <div id="taxFreeAmountDetail" class="section">
-                    <h2 class="flush--bottom">@Messages("tai.taxFreeAmount.detailsection.heading")</h2>
 
                     @taxFreeAmountSummary(viewModel.taxFreeAmountSummary.summaryItems)
 
-                </div>
+            </div>
+                    <br>
```
Comment (brendanstraughan) [inline app/views/TaxFreeAmountView.scala.html:80]: @masahmad are we allowed to use `<br>`

Comment (masahmad) [reply app/views/TaxFreeAmountView.scala.html:80]: good spot - yes we should take out the BR and use a bottom margin instead   - the accessibility team don't like anyone using BR's

File: app/views/includes/taxFreeAmountSummary.scala.html (hunk 4 above)
Comment (masahmad) [reply app/views/includes/taxFreeAmountSummary.scala.html:76]: yes needs removing

File: test/views/html/TaxFreeAmountViewSpec.scala (hunk 3 above)
Comment (masahmad) [reply test/views/html/TaxFreeAmountViewSpec.scala:40]: yes intentionally removed  as the figure has moved outside of this element

File: app/views/includes/taxFreeAmountSummary.scala.html (hunk 4 above)
Comment (masahmad) [reply app/views/includes/taxFreeAmountSummary.scala.html:76]: highlight  introdcued a grey shade in the amount row - which does not exist in prototype so its been removed - also there a corresponding test has been removed for the highlight.

File: test/views/html/TaxFreeAmountViewSpec.scala [hunk 6]
```diff
@@ -112,7 +112,7 @@ class TaxFreeAmountViewSpec extends TaiViewSpec {
       "visually formats the final table" when {
         "the corresponding final summary item view model contains only a single row" in {
           doc.select("#summaryTable4 .cya-question").size() mustBe 1
-          doc must haveElementAtPathWithClass("#summaryTable4 .govuk-check-your-answers", "highlight")
```
Comment (brendanstraughan) [inline test/views/html/TaxFreeAmountViewSpec.scala:115]: could we not still have the test just without the "highlight"

File: app/views/TaxFreeAmountView.scala.html (hunk 5 above)
Comment (brendanstraughan) [reply app/views/TaxFreeAmountView.scala.html]: does it still look ok without the br

**Review by brendanstraughan — APPROVED** [review]

**Pull request merged**